scripts/common.scala

```diff
@@ -8,11 +8,34 @@ import java.util.concurrent.atomic.AtomicInteger
 import scala.annotation.tailrec
 import scala.concurrent.duration.Duration
 
-def besomDir: os.Path =
-  if os.pwd.last != "besom" then
-    println("You have to run this command from besom project root directory")
-    sys.exit(1)
-  os.pwd
+import besom.codegen.Config
+
+object Args:
+  def parse(args: Seq[String], monoFlags: Seq[String] = Seq.empty): (Vector[String], Map[String, Int | String]) =
+    val (rest, flags) = parseInner(args, monoFlags)
+    val mergedFlags: Map[String, Int | String] = flags.groupMapReduce(_._1)(_._2) {
+      case (a: Int, b: Int)       => a + b
+      case (a: Int, b: String)    => a + Integer.parseInt(b)
+      case (a: String, b: Int)    => Integer.parseInt(a) + b
+      case (a: String, b: String) => b // last one wins
+    }
+    (rest, mergedFlags)
+
+  private def parseInner(args: Seq[String], monoFlags: Seq[String]): (Vector[String], Vector[(String, Int | String)]) =
+    args match
+      case Nil => (Vector.empty, Vector.empty)
+      case v +: tail if monoFlags.contains(s"--$v") || monoFlags.contains(s"-$v") =>
+        val (rest, flags) = parseInner(tail, monoFlags)
+        (rest, (v, 1) +: flags)
+      case s"--$name" +: value +: tail =>
+        val (rest, flags) = parseInner(tail, monoFlags)
+        (rest, (name, value) +: flags)
+      case s"-$name" +: value +: tail =>
+        val (rest, flags) = parseInner(tail, monoFlags)
+        (rest, (name, value) +: flags)
+      case head +: tail =>
+        val (rest, flags) = parseInner(tail, monoFlags)
+        (head +: rest, flags)
```

Comment:
at this point it wouldn't it make sense to use a proper cli parser library? case-app? scopt? scallop?

Reply:
I didn't want to over-engineer a scripts module :)